services/repository/src/main/java/org/sagebionetworks/auth/filter/AcceptTermsOfUseFilter.java

```diff
@@ -0,0 +1,82 @@
+package org.sagebionetworks.auth.filter;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.sagebionetworks.auth.HttpAuthUtil;
+import org.sagebionetworks.auth.services.AuthenticationService;
+import org.sagebionetworks.repo.model.AuthorizationConstants;
+import org.sagebionetworks.repo.model.AuthorizationConstants.BOOTSTRAP_PRINCIPAL;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+/**
+
+ * This filter checks that the authenticated Syanpse user has agreed to the Synapse Terms Of Use.
+ * Anonymous users are simply let through.
+ */
+@Component("acceptTermsOfUseFilter")
+public class AcceptTermsOfUseFilter implements Filter {
+	private static final String TOU_UNSIGNED_REASON = "Terms of use have not been signed.";
+	
+	private AuthenticationService authenticationService;
+
+	@Autowired
+	public AcceptTermsOfUseFilter(AuthenticationService authenticationService) {
+		this.authenticationService = authenticationService;
+	}
+	
+	@Override
+	public final void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)

```

Comment:
A little trickery in the (CGLIB) proxies that we use for the spring beans, I'm afraid that the method cannot be final